tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -276,56 +276,71 @@ class UpgradeIntegrationTest extends AkkaIntegrationTest with MesosClusterTest w
-    (marathon16549.client.info.entityJson \ "version").as[String] should be("1.6.549")
+      // Start apps in 1.6.549
+      Given("A Marathon 1.6.549 is running")
+      marathon16549.start().futureValue
+      (marathon16549.client.info.entityJson \ "version").as[String] should be("1.6.549")
 
-    And("new running apps in Marathon 1.6.549")
-    val containerPath = "persistent-volume"
-    val residentApp_16549 = residentApp(
-      id = testBasePath / "resident-app-16549",
-      containerPath = containerPath,
-      cmd = s"""echo $$MESOS_TASK_ID >> $containerPath/data && sleep 1000""")
-    marathon16549.client.createAppV2(residentApp_16549) should be(Created)
+      And("new running apps in Marathon 1.6.549")
+      val containerPath = "persistent-volume"
+      val residentApp_16549 = residentApp(
+        id = testBasePath / "resident-app-16549",
+        containerPath = containerPath,
+        cmd = s"""echo $$MESOS_TASK_ID >> $containerPath/data && sleep 13371337""")
+      marathon16549.client.createAppV2(residentApp_16549) should be(Created)
 
-    patienceConfig
-    eventually { marathon16549 should have (runningTasksFor(residentApp_16549.id.toPath, 1)) }
-    val originalApp16549Tasks = marathon16549.client.tasks(residentApp_16549.id.toPath).value
-
-    When("We restart the app")
-    marathon16549.client.restartApp(residentApp_16549.id.toPath) should be(OK)
+      patienceConfig
+      eventually { marathon16549 should have (runningTasksFor(residentApp_16549.id.toPath, 1)) }
+      val originalApp16549Tasks = marathon16549.client.tasks(residentApp_16549.id.toPath).value
 
-    Then("We have new running tasks")
-    eventually {
-      marathon16549.client.tasks(residentApp_16549.id.toPath).value should not contain theSameElementsAs(originalApp16549Tasks)
-      marathon16549 should have (runningTasksFor(residentApp_16549.id.toPath, 1))
-    }
+      When("We restart the app")
+      marathon16549.client.restartApp(residentApp_16549.id.toPath) should be(OK)
 
-    // Pass upgrade to current
-    When("Marathon is upgraded to the current version")
-    marathon16549.stop().futureValue
-    val marathonCurrent = LocalMarathon(suiteName = s"$suiteName-current", masterUrl = mesosMasterUrl, zkUrl = zkUrl)
-    marathonCurrent.start().futureValue
-    (marathonCurrent.client.info.entityJson \ "version").as[String] should be(BuildInfo.version.toString)
+      Then("We have new running tasks")
+      eventually {
+        marathon16549.client.tasks(residentApp_16549.id.toPath).value should not contain theSameElementsAs(originalApp16549Tasks)
+        marathon16549 should have (runningTasksFor(residentApp_16549.id.toPath, 1))
+      }
 
-    Then("All apps from 1.6.549 are still running")
-    marathonCurrent should have (runningTasksFor(residentApp_16549.id.toPath, 1))
-    val restartedApp16549Tasks = marathonCurrent.client.tasks(residentApp_16549.id.toPath).value
+      // ==== TODO(ad): REMOVE ME AFTERWARDS =====
```

Comment:
Not sure what this todo means. After when should this be removed?

Reply:
After I found out why this test is flaky.